src/univariate/continuous/skewexponentialpower.jl

```diff
@@ -0,0 +1,115 @@
+SkewExponentialPower(μ, σ)        # SEPD with location μ, scale σ, shape 2, and skewness 0.5 (the normal distribution)
+SkewExponentialPower(μ)           # SEPD with location μ, scale 1, shape 2, and skewness 0.5 (the normal distribution)
+
+params(d)       # Get the parameters, i.e. (μ, σ, p, α)
+shape(d)        # Get the shape parameter, i.e. p
+location(d)     # Get the location parameter, i.e. μ
+scale(d)        # Get the scale parameter, i.e. σ
+```
+"""
+struct SkewExponentialPower{T <: Real} <: ContinuousUnivariateDistribution
+    μ::T
+    σ::T
+    p::T
+    α::T
+    SkewExponentialPower{T}(μ::T, σ::T, p::T, α::T) where {T} = new{T}(μ, σ, p, α)
+end
+
+function SkewExponentialPower(µ::T, σ::T, p::T, α::T; check_args=true) where {T <: Real}
+    check_args && @check_args(SkewExponentialPower, σ > zero(σ))
+    check_args && @check_args(SkewExponentialPower, p > zero(p))
+    check_args && @check_args(SkewExponentialPower, α > zero(α) && α < one(α))
+    return SkewExponentialPower{T}(µ, σ, p, α)
+end
+
+SkewExponentialPower(μ::Real, σ::Real, p::Real, α::Real) = SkewExponentialPower(promote(μ, σ, p, α)...)
+SkewExponentialPower(μ::Real, σ::Real, p::Real) = SkewExponentialPower(promote(μ, σ, p, 0.5)...)
+SkewExponentialPower(μ::Real, σ::Real) = SkewExponentialPower(promote(μ, σ, 2., 0.5)...)
+SkewExponentialPower(μ::Real) = SkewExponentialPower(promote(μ, 1., 2., 0.5)...)
+SkewExponentialPower() = SkewExponentialPower(0., 1., 2., 0.5)
+
+@distr_support SkewExponentialPower -Inf Inf
+
+### Conversions
+convert(::Type{SkewExponentialPower{T}}, μ::S, σ::S, p::S, α::S) where {T <: Real, S <: Real} = SkewExponentialPower(T(μ), T(σ), T(p), T(α))
+convert(::Type{SkewExponentialPower{T}}, d::SkewExponentialPower{S}) where {T <: Real, S <: Real} = SkewExponentialPower(T(d.μ), T(d.σ), T(d.p), T(d.α), check_args=false)
+
+### Parameters
+@inline partype(d::SkewExponentialPower{T}) where {T<:Real} = T
+
+params(d::SkewExponentialPower) = (d.μ, d.σ, d.p, d.α)
+location(d::SkewExponentialPower) = d.μ
+shape(d::SkewExponentialPower) = d.p
+scale(d::SkewExponentialPower) = d.σ
+
+### Statistics
+
+#Calculates the kth central moment of the SEPD, see Equation 18 in [1].
+function m_k(d::SkewExponentialPower, k::Integer)
+    _, σ, p, α = params(d)
+    (2*p^(1/p))^k*((-1)^k*α^(1+k) + (1-α)^(1+k)) *
+        σ^k * gamma((1+k)/p) / gamma(1/p)
+end
+
+mean(d::SkewExponentialPower) = m_k(d, 1) + d.μ
+mode(d::SkewExponentialPower) =  m_k(d, 1) + d.μ
+var(d::SkewExponentialPower) = m_k(d, 2) - m_k(d, 1)^2
+std(d::SkewExponentialPower) = √var(d)
+skewness(d::SkewExponentialPower) = m_k(d, 3)/(std(d))^3
+kurtosis(d::SkewExponentialPower) = m_k(d, 4)/var(d)^2 - 3.
+
+function logpdf(d::SkewExponentialPower, x::Real)
+    μ, σ, p, α = params(d)
+    -log(σ) - log(2*p^(1/p)*gamma(1+1/p)) - 1/p*(x < μ ? ((μ-x)/(2*σ*α))^p : ((x-μ)/(2*σ*(1-α)))^p)
```

Comment:
Again, I think it would be better to perform more calculations in log space and use `loggamma`:
```suggestion
    a = x < μ ? α : 1 - α
    inv_p = inv(p)
    return -(logtwo + log(σ) + inv_p * log(p) + loggamma(1 + inv_p) + inv_p * (abs(μ - x) / (2 * σ * a))^p)
```

Reply:
Changed to use `loggamma`